ui/src/criteria/concept.tsx

```diff
@@ -80,22 +87,26 @@ function ConceptEdit(props: ConceptEditProps) {
         (res) => {
           if (res.instances) {
             setRows(
-              res.instances.map((instance, i) => {
-                const row: Row = { id: i };
-                for (const k in instance) {
-                  const v = instance[k];
-                  if (!v) {
-                    row[k] = "";
-                  } else if (!!v.int64Val) {
-                    row[k] = v.int64Val;
-                  } else if (!!v.boolVal) {
-                    row[k] = v.boolVal;
-                  } else {
-                    row[k] = v.stringVal;
+              res.instances
+                .map((instance) => {
+                  const row: Row = {
+                    id: instance["concept_id"]?.int64Val || 0,
+                  };
+                  for (const k in instance) {
+                    const v = instance[k];
+                    if (!v) {
+                      row[k] = "";
+                    } else if (v.int64Val !== null) {
+                      row[k] = v.int64Val;
+                    } else if (v.boolVal !== null) {
+                      row[k] = v.boolVal;
+                    } else {
+                      row[k] = v.stringVal;
+                    }
                   }
-                }
-                return row;
-              })
+                  return row;
+                })
+                .filter((row) => row.id != 0)
```

Comment:
nit: I recommend using === or !== only, the truth table for == is mind-bending: https://dorey.github.io/JavaScript-Equality-Table/